### PR TITLE
Populate datastore fields on vicadmin portal

### LINKED ISSUE #1939: vicadmin reports incorrect storage availability
Build 3910 (8/11/16)

installed on VC but pointed to specific datastore on host for image-store.

Install command:
./vic-machine-darwin create --target 10.161.25.72 --image-store ds1 --name vic-test-mike3 --user Administrator@vsphere.local --password 'Admin!23'  --compute-resource /ha-datacenter/host/10.161.236.25/Resources --bridge-network bridge --external-network vm-network

Datastore ds1 has around 23.06GB free. 

VIC admin page reports:
Virtual Container Host Info
Docker Endpoint: 10.161.230.180:2376
Storage: 420GB Remaining

### PR DESCRIPTION
**Pull request by kjplatz**
Populate the storage remaining field on the datastore in the vicadmin portal.
Fixes #1939

### CODE REVIEW

File: lib/vicadmin/validate.go [hunk 1]
```diff
@@ -130,5 +134,23 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 		v.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
 	}
 
+	v.QueryDatastore(ctx, vch, sess)
 	return v
 }
+
+func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualContainerHostConfigSpec, sess *session.Session) {
+	var ds []mo.Datastore
+	refs := []types.ManagedObjectReference{sess.Datastore.Reference()}
```
Comment (emlin) [inline lib/vicadmin/validate.go]: where did you get this datastore? from image datastore or volume datastore?

**Comment by caglar10ur** [comment]
LGTM :sheep: :it: (title says datstore :P)

File: lib/vicadmin/validate.go (hunk 1 above)
Comment (kjplatz) [inline lib/vicadmin/validate.go]: I got that datastore from the session variable, which I assume is (currently) the image datastore[0].  I'm revising to pull all datastores (including datastores which contain volumes)

Comment (kjplatz) [inline lib/vicadmin/validate.go]: It now pulls in information from all defined datastores and volume stores.

File: lib/vicadmin/validate.go [hunk 2]
```diff
@@ -138,18 +139,57 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
...
+	sort.Strings(dsNames)
+	lastDs := ""
+
+	refs := []types.ManagedObjectReference{}
+	for _, dsName := range dsNames {
+		if lastDs == dsName {
+			continue
+		}
+		lastDs = dsName
+		ds, err := sess.Finder.DatastoreOrDefault(ctx, dsName)
```
Comment (emlin) [inline lib/vicadmin/validate.go]: Should use Finder.Datastore method. You don't need to use default datastore if the dsName is ""

**Comment by emlin** [comment]
cool. Just one minor comment. Everything else LGTM. 👍

**Pull request merged**